torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -756,7 +756,7 @@ std::shared_ptr<ProcessGroupNCCL::WorkNCCL> ProcessGroupNCCL::initWork(
 
 c10::intrusive_ptr<c10::ivalue::Future> ProcessGroupNCCL::WorkNCCL::
     getFuture() {
-  return futureWork_;
+  return c10::make_intrusive<FutureNCCL>(shared_from_this(), outputs_);
```

Comment:
nit: Add a comment here about the circular reference and why we need to create a new object here.